deps/build.jl

```diff
@@ -1,285 +1,78 @@
-    "MPI_SUM",
-
-    # request
-    "MPI_REQUEST_NULL",
-
-    # info
-    "MPI_MAX_INFO_KEY",
-    "MPI_MAX_INFO_VAL",
-    "MPI_INFO_NULL",
-
-
-    # proc
-    "MPI_PROC_NULL",
-
-    # memory layout
-    "MPI_ORDER_C",
-    "MPI_ORDER_FORTRAN",
-
-    # status
-    "MPI_STATUS_SIZE", # size of Status array
-    "MPI_STATUS_IGNORE",
-    "MPI_SOURCE", # index in Status array
-    "MPI_TAG",    # index in Status array
-    "MPI_ERROR",  # index in Status array
-
-    # lock_type
-    "MPI_LOCK_EXCLUSIVE",
-    "MPI_LOCK_SHARED",
-
-    # window
-    "MPI_WIN_NULL",
-
-    # misc constants
-    "MPI_ANY_SOURCE",
-    "MPI_ANY_TAG",
-    "MPI_TAG_UB", # upper bound for tag value
-    "MPI_UNDEFINED",
-]
-
-MPI_functions = [
-    "MPI_ABORT",
-    "MPI_ACCUMULATE",
-    "MPI_ALLGATHER",
-    "MPI_ALLGATHERV",
-    "MPI_ALLREDUCE",
-    "MPI_ALLTOALL",
-    "MPI_ALLTOALLV",
-    "MPI_ALLTOALLW",
-    "MPI_BARRIER",
-    "MPI_BCAST",
-    "MPI_BSEND",
-    "MPI_CANCEL",
-    "MPI_CART_CREATE",
-    "MPI_CART_COORDS",
-    "MPI_CART_SHIFT",
-    "MPI_COMM_DUP",
-    "MPI_COMM_FREE",
-    "MPI_COMM_GET_PARENT",
-    "MPI_COMM_RANK",
-    "MPI_COMM_SIZE",
-    "MPI_COMM_SPLIT",
-    "MPI_COMM_SPLIT_TYPE",
-    "MPI_DIMS_CREATE",
-    "MPI_EXSCAN",
-    "MPI_FETCH_AND_OP",
-    "MPI_FINALIZE",
-    "MPI_FINALIZED",
-    "MPI_GATHER",
-    "MPI_GATHERV",
-    "MPI_GET",
-    "MPI_GET_ACCUMULATE",
-    "MPI_GET_ADDRESS",
-    "MPI_GET_COUNT",
-    "MPI_GET_PROCESSOR_NAME",
-    "MPI_IALLTOALL",
-    "MPI_IALLTOALLV",
-    "MPI_IALLTOALLW",
-    "MPI_INFO_CREATE",
-    "MPI_INFO_DELETE",
-    "MPI_INFO_FREE",
-    "MPI_INFO_GET",
-    "MPI_INFO_GET_NKEYS",
-    "MPI_INFO_GET_NTHKEY",
-    "MPI_INFO_GET_VALUELEN",
-    "MPI_INFO_SET",
-    "MPI_INIT",
-    "MPI_INITIALIZED",
-    "MPI_INTERCOMM_MERGE",
-    "MPI_IPROBE",
-    "MPI_IRECV",
-    "MPI_ISEND",
-    "MPI_OP_CREATE",
-    "MPI_OP_FREE",
-    "MPI_PACK",
-    "MPI_PACK_SIZE",
-    "MPI_PROBE",
-    "MPI_PUT",
-    "MPI_RECV",
-    "MPI_RECV_INIT",
-    "MPI_REDUCE",
-    "MPI_REQUEST_FREE",
-    "MPI_RSEND",
-    "MPI_SCAN",
-    "MPI_SCATTER",
-    "MPI_SCATTERV",
-    "MPI_SEND",
-    "MPI_SEND_INIT",
-    "MPI_SSEND",
-    "MPI_TEST",
-    "MPI_TESTALL",
-    "MPI_TESTANY",
-    "MPI_TESTSOME",
-    "MPI_TYPE_CONTIGUOUS",
-    "MPI_TYPE_CREATE_STRUCT",
-    "MPI_TYPE_CREATE_SUBARRAY",
-    "MPI_TYPE_COMMIT",
-    "MPI_UNPACK",
-    "MPI_WAIT",
-    "MPI_WAITALL",
-    "MPI_WAITANY",
-    "MPI_WAITSOME",
-    "MPI_WIN_ATTACH",
-    "MPI_WIN_ALLOCATE_SHARED",
-    "MPI_WIN_CREATE",
-    "MPI_WIN_CREATE_DYNAMIC",
-    "MPI_WIN_DETACH",
-    "MPI_WIN_FENCE",
-    "MPI_WIN_FLUSH",
-    "MPI_WIN_FREE",
-    "MPI_WIN_LOCK",
-    "MPI_WIN_SHARED_QUERY",
-    "MPI_WIN_SYNC",
-    "MPI_WIN_UNLOCK",
-    "MPI_WTICK",
-    "MPI_WTIME",
-    ]
-
-MPI_C_constants = [
-    "MPI_IN_PLACE",
-]
-MPI_C_types = ["Comm", "Info", "Win"]
-
+const libmpi = find_library(Sys.iswindows() ? "msmpi.dll" : "libmpi",
```

Comment:
`find_library` checks the provided paths, `DL_LOAD_PATH`, and various system paths. On supercomputers -- as an example see Summit below -- I have rarely seen `DL_LOAD_PATH` being set.
We need a better error message that informs the user can set `JULIA_MPI_PATH`. Alternatively `which mpiexec` might work

```
whatis("Name : spectrum-mpi")
whatis("Version : 10.3.0.0-20190419")
whatis("Short description : IBM MPI implementation from Spectrum MPI.")
help([[IBM MPI implementation from Spectrum MPI.]])
family("mpi")
prepend_path("MODULEPATH","/autofs/nccs-svm1_sw/summit/modulefiles/site/linux-rhel7-ppc64le/spectrum-mpi/10.3.0.0-20190419-4um5hjo/gcc/6.4.0")
setenv("LMOD_MPI_NAME","spectrum-mpi")
setenv("LMOD_MPI_VERSION","10.3.0.0-20190419-4um5hjo")
prepend_path("PATH","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/bin")
prepend_path("MANPATH","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/share/man")
prepend_path("LD_LIBRARY_PATH","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/lib")
prepend_path("LIBRARY_PATH","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/lib")
prepend_path("CPATH","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/include")
prepend_path("CMAKE_PREFIX_PATH","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/")
setenv("OMPI_DIR","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6")
setenv("MPI_ROOT","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6")
setenv("OPAL_PREFIX","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6")
setenv("OPAL_LIBDIR","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6/lib")
setenv("OMPI_CC","gcc")
setenv("OMPI_CXX","g++")
setenv("OMPI_FC","gfortran")
setenv("OLCF_SPECTRUM_MPI_ROOT","/autofs/nccs-svm1_sw/summit/.swci/1-compute/opt/spack/20180914/linux-rhel7-ppc64le/gcc-6.4.0/spectrum-mpi-10.3.0.0-20190419-4um5hjogm3tepg4xe23hrptlrs2y7ez6")
setenv("PAMI_IBV_ENABLE_OOO_AR","1")
setenv("PAMI_IBV_QP_SERVICE_LEVEL","8")
```

Reply:
if you set `JULIA_MPI_PATH` does it work?

Reply:
or do you need to `module load XXmpiXX`?